sdk/src/test/java/com/google/cloud/dataflow/sdk/util/common/CounterSetTest.java

```diff
@@ -222,4 +226,110 @@ public void testAddCounterMutatorAddMultipleCounters() {
     assertThat(set, containsInAnyOrder(c1, c2));
   }
 
+  @Test
+  public void testHandlesDuplicateEntries() {
+    final String counterName = "counter-name";
+    CounterSet counters = new CounterSet();
+
+    // getting a nonexistent counter returns null
+    assertThat(counters.getExistingCounter(counterName), is(nullValue()));
+
+    // adding an already existing counter using addNewCounter throws an exception
+    Counter<Integer> counterAddCheck = Counter.ints(counterName, SUM);
+    boolean exceptionThrown = false;
+    try {
+      counters.addNewCounter(counterAddCheck);
+    } catch (IllegalArgumentException e) {
+      exceptionThrown = true;
+    }
+    assertThat(exceptionThrown, is(false));
+
+    try {
+      counters.addNewCounter(counterAddCheck);
+    } catch (IllegalArgumentException e) {
+      exceptionThrown = true;
+    }
+    assertThat(exceptionThrown, is(true));
+    assertThat(counters.getExistingCounter(counterName), is(notNullValue()));
+  }
+
+  @Test
+  public void testMutatorSanity() {
```

Comment:
Better name and/or add a comment on what constitutes "sanity". Also, suggest splitting this up into more tests for specific cases.